workflow/controller/dag.go

```diff
@@ -314,6 +314,14 @@ func (woc *wfOperationCtx) executeDAGTask(dagCtx *dagContext, taskName string) {
 	node := dagCtx.getTaskNode(taskName)
 	task := dagCtx.GetTask(taskName)
 	if node != nil && node.Fulfilled() {
+		// Collect the completed task metrics
+		tmpl := woc.wf.GetTemplateByName(task.Template)
+		if tmpl != nil && tmpl.Metrics != nil {
+			if prevNodeStatus, ok := woc.preExecutionNodePhases[node.ID]; ok && !prevNodeStatus.Fulfilled() {
+				localScope, realTimeScope := woc.prepareMetricScope(node)
+				woc.computeMetrics(tmpl.Metrics.Prometheus, localScope, realTimeScope, false)
+			}
+		}
```

Comment:
Not too sure this is the correct approach here. All metrics should be emitted within `executeTemplate`. If you notice for DAG or Steps, metrics are never emitted outside of it. Can you investigate why the current metric emission code in `executeTemplate` is not getting called in this case?

Reply:
In DAG flow, the Code is checking the `node.Fulfilled()` in  early-stage  (executeDAGTask) which prevents to execute the `executeTemplate` for completed the tasks.  
https://github.com/argoproj/argo/blob/5b5d2359ef9f573121fe6429e386f03dd8652ece/workflow/controller/dag.go#L316-L326

Options:
1.  We can remove the `return` statement on the above code. So, Completed node will travel to `executetemplate` But I am not sure whether it will break ay DAG flow
2. Above fix, Add metrics emission code in `node.Fulfilled()` check

Reply:
On second thought, I think we can keep this current fix for this issue.